lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -500,6 +501,12 @@ void OCLBackend::executeConvolution(const OCLConvolutionInst *CC) {
                                 ((M_FW_ - 1) / fw_div_M + 1) * fw_wgs1,
                                 idim.n * group};
 
+  // The global work for the N dimension (which covers all pixels) should be big
+  // enough to cover all outputs for the correctness.
+  if (global[0] * fw_div_N < odim.h * odim.w) {
+    global[0] = (odim.h * odim.w - 1) / fw_div_N + 1;
```

Comment:
that's the same as (w * h + fw_div_n - 1) / fwdivN :P, but looks good.

Reply:
Yes, I just converted it from your expression :)